FinSearchUnified/Tests/Bundle/SearchBundleFindologic/FacetHandler/ImageFacetHandlerTest.php

```diff
@@ -0,0 +1,203 @@
+     * @param ConditionInterface|null $condition
+     * @param array $facetData
+     */
+    public function testGeneratesPartialFacetBasedOnFilterDataAndActiveConditions(
+        array $filterData,
+        ConditionInterface $condition = null,
+        array $facetData = []
+    ) {
+        $facet = new ProductAttributeFacet(
+            'vendor',
+            ProductAttributeFacet::MODE_VALUE_LIST_RESULT,
+            'vendor',
+            'Manufacturer'
+        );
+
+        $active = false;
+        $criteria = new Criteria();
+
+        if ($condition !== null) {
+            $active = true;
+            $criteria->addCondition($condition);
+        }
+
+        $filter = $this->generateFilter($filterData);
+
+        $responses = [];
+
+        foreach ($filterData as $filterDatum) {
+            if ($filterDatum['image']) {
+                $responses[] = new Response($filterDatum['status']);
+            }
+        }
+
+        $client = new Client();
+
+        // Create a mock subscriber and queue responses.
+        $mock = new Mock($responses);
+
+        // Add the mock subscriber to the client.
+        $client->getEmitter()->attach($mock);
+
+        $guzzleFactoryMock = $this->createMock(GuzzleFactory::class);
+        $guzzleFactoryMock->method('createClient')->willReturn($client);
+
+        $facetHandler = new ImageFacetHandler($guzzleFactoryMock, []);
+
+        $result = $facetHandler->generatePartialFacet($facet, $criteria, $filter);
+
+        $mediaListItems = [];
+
+        foreach ($facetData as $facetDatum) {
+            if ($facetDatum[2] !== null) {
+                $media = new Media();
+                $media->setFile($facetDatum[2]);
+            } else {
+                $media = null;
+            }
+
+            $mediaItem = new MediaListItem($facetDatum[0], $facetDatum[0], $facetDatum[1], $media);
+            $mediaListItems[] = $mediaItem;
+        }
+
+        $facetResult = new MediaListFacetResult(
+            'product_attribute_vendor',
+            $active,
+            'Manufacturer',
+            $mediaListItems,
+            'vendor'
+        );
+
+        $this->assertEquals($facetResult, $result);
+    }
+
+    public function imageFilterProvider()
+    {
+        return [
+            'Image filter with condition' => [
+                [
+                    ['name' => 'Red', 'image' => '', 'status' => 404],
+                    ['name' => 'Green', 'image' => '', 'status' => 404],
+                    ['name' => 'Zima Blue', 'image' => 'https://example.com/zima-blue.gif', 'status' => 200],
+                    ['name' => 'Yellow', 'image' => 'https://example.com/yellow.gif', 'status' => 200],
+                    ['name' => 'Purple', 'image' => 'https://example.com/purple.gif', 'status' => 404],
+                    ['name' => 'Light Purple', 'image' => 'https://example.com/light-purple.gif', 'status' => 404],
+                ],
+                // Condition
```

Comment:
Instead of adding comments, use string keys for the arrays (should be named like the respective argument).